### ISSUE TITLE
trtserver: regionFormat.cpp:65: size_t nvinfer1::RegionFormatB::memorySize(int, const nvinfer1::Dims&) const: Assertion `batchSize > 0' failed.

### ISSUE THREAD

**Issue by zhangqijun** [NVIDIA/tensorrt-inference-server#550]
when I use tf-trt int8 optimization model,with nvcr.io/nvidia/tensorrtserver:19.07-py3.
I got this error,and server down. sorry about my Chinglish.

> 
2019-08-13 06:52:23.686850: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:525] layout failed: Invalid argument: The graph is already optimized by layout optimizer.
2019-08-13 06:52:24.241588: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:632] Building a new TensorRT engine for TRTEngineOp_0 input shapes: [[8,112,112,3]]
2019-08-13 06:52:24.241834: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  GridAnchor_TRT Namespace 
2019-08-13 06:52:24.241842: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  NMS_TRT Namespace 
2019-08-13 06:52:24.241846: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Reorg_TRT Namespace 
2019-08-13 06:52:24.241849: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Region_TRT Namespace 
2019-08-13 06:52:24.241875: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Clip_TRT Namespace 
2019-08-13 06:52:24.241878: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  LReLU_TRT Namespace 
2019-08-13 06:52:24.241881: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  PriorBox_TRT Namespace 
2019-08-13 06:52:24.241884: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Normalize_TRT Namespace 
2019-08-13 06:52:24.241887: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  RPROI_TRT Namespace 
2019-08-13 06:52:24.241890: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  BatchedNMS_TRT Namespace 
2019-08-13 06:52:24.241893: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  GridAnchor_TRT Namespace 
2019-08-13 06:52:24.241896: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  NMS_TRT Namespace 
2019-08-13 06:52:24.241912: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Reorg_TRT Namespace 
2019-08-13 06:52:24.241917: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Region_TRT Namespace 
2019-08-13 06:52:24.241919: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Clip_TRT Namespace 
2019-08-13 06:52:24.241937: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  LReLU_TRT Namespace 
2019-08-13 06:52:24.241941: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  PriorBox_TRT Namespace 
2019-08-13 06:52:24.241944: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  Normalize_TRT Namespace 
2019-08-13 06:52:24.241948: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  RPROI_TRT Namespace 
2019-08-13 06:52:24.241951: E tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:41] DefaultLogger Could not register plugin creator  BatchedNMS_TRT Namespace 
2019-08-13 06:52:24.241955: E tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:1097] Failed to initialize TensorRT plugins, and conversion may fail later.
trtserver: regionFormat.cpp:65: size_t nvinfer1::RegionFormatB::memorySize(int, const nvinfer1::Dims&) const: Assertion `batchSize > 0' failed.
> 

pb model is generated by this code:
`with graph.as_default():
    with tf.Session(config=config) as sess:
        # First deserialize your frozen graph:
        with tf.gfile.GFile("arcface.pb", "rb") as f:
            graph_def = tf.GraphDef()
            graph_def.ParseFromString(f.read())
        # Now you can create a TensorRT inference graph from your
        # frozen graph:
        trt_graph = trt.create_inference_graph(
            input_graph_def=graph_def,
            outputs=['embd_extractor/BatchNorm_1/Reshape_1'],
            max_batch_size=8,
            max_workspace_size_bytes=2 << 20,
            precision_mode="int8"
            )
import cv2
with tf.Session(graph=tf.Graph(),config=config) as sess:
   output_node = tf.import_graph_def(trt_graph, return_elements=['embd_extractor/BatchNorm_1/Reshape_1'])[0]
   images = tf.get_default_graph().get_tensor_by_name("import/Placeholder:0")
   output_tenser = tf.get_default_graph().get_tensor_by_name("import/embd_extractor/BatchNorm_1/Reshape_1:0")
   df = pd.read_csv("/media/ssd/predev/face_verifacation/dairy_test/dairy_verifacation_align.csv")
   for i in range(len(df)):
       img_path = os.path.join("/media/ssd/predev/face_verifacation/dairy_test/images_align/img",df.loc[i,"A1"])
       img = cv2.imread(img_path).astype(np.float32)[:,:,::-1]
       img /= 127.5
       img -= 1
       print(sess.run(output_tenser,feed_dict={images:[img]}).shape)

   trt_int8_calibrated_graph = trt.calib_graph_to_infer_graph(trt_graph,is_dynamic_op=True)
   tf.train.write_graph(trt_int8_calibrated_graph, './', 'int8model.graphdef', as_text=False)`

**Comment by deadeyegoodwin** [comment]
What does you model configuration look like (config.pbtxt).

**Comment by zhangqijun** [comment]
this is config.pbtxt


name: "arcfaceint8"
platform: "tensorflow_graphdef"
max_batch_size: 16
input [
   {
      name: "Placeholder"
      data_type: TYPE_FP32
      dims: [ 112, 112 ,3]
   }
]
output [
   {
      name: "embd_extractor/BatchNorm_1/Reshape_1"
      data_type: TYPE_FP32
      dims: [ 512 ]
   }
]
dynamic_batching {
  preferred_batch_size: [ 8, 16 ]
  max_queue_delay_microseconds: 100
}

**Comment by deadeyegoodwin** [comment]
Did you generate the TF-TRT model using the same version of TensorRT as is being used by TRTIS? The easiest way to do this is to use the TensorRT container from the same release as the TRTIS you are using  (for example, 19.07).

**Comment by minhdeal** [comment]
Any updates on this issue? I have the same issue on  tensorflow/serving:1.14.0-gpu and nvcr.io/nvidia/tensorflow:19.10-py3.

**Comment by helenHlz** [comment]
@zhangqijun Sorry to bother you, do you have a solution to this problem?

**Comment by helenHlz** [comment]
@minhdeal  Sorry to bother you, do you have a solution to this problem?
